cmd/restic/cmd_ls.go

```diff
@@ -64,25 +62,14 @@ func init() {
 
 type lsSnapshot struct {
 	*restic.Snapshot
-
 	ID         *restic.ID `json:"id"`
 	ShortID    string     `json:"short_id"`
-	Nodes      []lsNode   `json:"nodes"`
 	StructType string     `json:"struct_type"` // "snapshot"
 }
 
 type lsNode struct {
-	Name       string      `json:"name"`
-	Type       string      `json:"type"`
-	Path       string      `json:"path"`
-	UID        uint32      `json:"uid"`
-	GID        uint32      `json:"gid"`
-	Size       uint64      `json:"size,omitempty"`
-	Mode       os.FileMode `json:"mode,omitempty"`
-	ModTime    time.Time   `json:"mtime,omitempty"`
-	AccessTime time.Time   `json:"atime,omitempty"`
-	ChangeTime time.Time   `json:"ctime,omitempty"`
-	StructType string      `json:"struct_type"` // "node"
+	*restic.Node
```

Comment:
We have created a different structure on purpose, to make it clear which fields are exported via `find` or `ls`, instead of the raw data from the repository. restic has two "contracts" (in terms of data structures) with the outside world: the repository (`restic.Node`) and the output of `find` and `ls`. Maybe we can reuse/refactor the code in `find` so that we'll always get the same output:

https://github.com/restic/restic/blob/d708d607fab766121d06ac64af55362daeb52a60/cmd/restic/cmd_find.go#L100-L113